crates/sui-indexer-alt-framework/src/lib.rs

```diff
@@ -41,23 +41,37 @@ pub mod mocks;
 /// Command-line arguments for the indexer
 #[derive(clap::Args, Default, Debug, Clone)]
 pub struct IndexerArgs {
-    /// Override for the checkpoint to start ingestion from -- useful for backfills. Otherwise, by
-    /// default, ingestion will start just after the lowest checkpoint watermark across all active
-    /// pipelines.
+    /// An optional task name for this indexer. When set, pipelines will record watermarks using the
+    /// format `{pipeline}{delimiter}{task}`. This allows the same pipelines to run under multiple
+    /// indexers (e.g. for backfills or temporary workflows) while maintaining separate watermark
+    /// entries in the database.
     ///
-    /// For both concurrent and sequential pipelines, if a first checkpoint is configured, and a
-    /// watermark does not exist for the pipeline, the indexer will also tell the pipeline to start
-    /// from this value.
+    /// By default there is no task name, and watermarks are keyed only by `pipeline`.
     ///
-    /// Unless `--skip-watermark` is set, this value must be less than or equal to the global high
-    /// watermark (preventing the indexer from introducing a gap in the data). This exception only
-    /// applies to concurrent pipelines, and these pipelines will also not report watermark updates.
+    /// Sequential pipelines cannot be attached to a tasked indexer.
     ///
-    /// Sequential pipelines will always start committing from the next checkpoint after its
-    /// watermark.
+    /// The framework ensures that tasked pipelines never commit checkpoints below the main
+    /// pipeline’s reader watermark. If pruning is enabled for the main pipeline, the tasked
+    /// pipeline should use the same pruning configuration to correctly track both the main
+    /// pipeline’s pruning and reader watermarks.
+    #[arg(long, requires = "delimiter")]
+    pub task: Option<String>,
+
+    #[arg(long, requires = "task")]
+    pub delimiter: Option<String>,
```

Comment:
I'm not following what this achieves -- this is the same as just having a single `task` flag, right? My earlier comment about requiring a delimiter was to say that the indexing framework should pick a delimiter character and enforce a property that pipeline names should not include that delimiter character.